### PR TITLE
#206 RqLengthAware doesn't intercept the other two read() methods

### PR DESCRIPTION
**Pull request by dmzaytsev**
for #206 
* implemented  the other two InputStream.read() methods

### CODE REVIEW

**Comment by davvd** [comment]
@dmzaytsev Thanks, someone will review your pull request soon

**Comment by davvd** [comment]
@longtimeago please review,thanks

**Comment by longtimeago** [comment]
@dmzaytsev basically, PR is fine, but no tests at all. Let's create several

**Comment by dmzaytsev** [comment]
@longtimeago look at the changes please

File: src/test/java/org/takes/rq/RqLengthAwareTest.java [hunk 1]
```diff
@@ -81,4 +83,89 @@ public void addsBigLengthToBody() throws IOException {
...
+     */
+    @Test
+    public void readsByteArray() throws IOException {
+        final String data = "array";
+        final InputStream stream = new RqLengthAware(
+            new RqFake(
+                Arrays.asList(
+                    "GET /test2",
+                    "Host: c.example.com",
+                    "Content-type: text/csv"
```
Comment (longtimeago) [inline src/test/java/org/takes/rq/RqLengthAwareTest.java]: do we need to specify content type here?

File: src/test/java/org/takes/rq/RqLengthAwareTest.java [hunk 2]
```diff
@@ -81,4 +83,89 @@ public void addsBigLengthToBody() throws IOException {
...
+    @Test
+    public void readsPartialArray() throws IOException {
+        final String data = "hello world";
+        final int len = 3;
+        final InputStream stream = new RqLengthAware(
+            new RqFake(
+                Arrays.asList(
+                    "GET /test3",
+                    "Host: d.example.com",
+                    "Content-type: text/plain; charset=us-ascii"
```
Comment (longtimeago) [inline src/test/java/org/takes/rq/RqLengthAwareTest.java]: same here

File: src/test/java/org/takes/rq/RqLengthAwareTest.java [hunk 3]
```diff
@@ -81,4 +83,89 @@ public void addsBigLengthToBody() throws IOException {
...
+     */
+    @Test
+    public void readsByte() throws IOException {
+        final String data = "{test}";
+        final InputStream stream = new RqLengthAware(
+            new RqFake(
+                Arrays.asList(
+                    "GET /test1",
+                    "Host: b.example.com",
+                    "Content-type: text/json"
```
Comment (longtimeago) [inline src/test/java/org/takes/rq/RqLengthAwareTest.java]: do we need to specify content type here?

**Comment by longtimeago** [comment]
@dmzaytsev one more remark

File: src/test/java/org/takes/rq/RqLengthAwareTest.java (hunk 3 above)
Comment (dmzaytsev) [inline src/test/java/org/takes/rq/RqLengthAwareTest.java]: @longtimeago According  the HTTP 1.1 specification any message containing an entity-body should include a Content-Type header. But  we can omit it here.  You're right

**Comment by dmzaytsev** [comment]
@longtimeago a Content-Type header removed
 take a look please

**Comment by longtimeago** [comment]
@rultor merge please

**Comment by yegor256** [comment]
@rultor merge pls

**Comment by davvd** [comment]
@elenavolokhova please, review this ticket for compliance with our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@davvd This one looks good.

**Comment by davvd** [comment]
> @davvd This one looks good.

@elenavolokhova thanks for the review and a good remark

**Comment by davvd** [comment]
@longtimeago 10 mins was added to the account of @elenavolokhova (for QA review), in transaction 56415045; **18 mins** was added to your account, many thanks for your contribution! The task took 56415057.; extra minutes for review comments (c=3); +18 added to your rating, at the moment it is: [+4776](http://www.netbout.com/b/35024?open=rating)

**Pull request merged**